### PR TITLE
add support for running locally without authentication

### PR DESCRIPTION
**Pull request by brpins**
## Type of change 🧩

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Documentation update
- [ ] Other (please explain in the description section above)

## Checklist before requesting a review

- [ ] I have performed a self-review of my own code
- [ ] I have double checked this work does not include any hardcoded secrets or passwords
- [ ] I have made corresponding changes to the documentation
- [ ] I have provided details on how I have tested my code
- [ ] I have referenced the ticket number above
- [ ] I have provided a description of the ticket
- [ ] I have included unit tests to cover any testable code changes

### CODE REVIEW

File: apps/web/.env.development [hunk 1]
```diff
@@ -33,12 +33,11 @@ API_HOST=http://localhost:3000
 
 # - - - - - - - - Auth - - - - - - - -
 
-AUTH_CLIENT_ID=<value goes here>
```
Comment (radum) [inline apps/web/.env.development:36]: I wouldn't remove these. Just so a dev knows what will need to be set in his .env.local to make it work.

Also if we are here, would you be so kinds to add a line in the readme explaining that for auth to work locally your newly env var needs to be turned on, and that these vars need to be set into .env.local?

File: apps/web/.env.development [hunk 2]
```diff
@@ -33,12 +33,11 @@ API_HOST=http://localhost:3000
...
-AUTH_CLOUD_INSTANCE_ID=<value goes here>
-AUTH_TENANT_ID=<value goes here>
-AUTH_CLIENT_SECRET=<value goes here>
-AUTH_REDIRECT_URI=<value goes here>
-
-AUTH_INSPECTOR_GROUP_ID=<value goes here>
-AUTH_CASEOFFICER_GROUP_ID=<value goes here>
-AUTH_VALIDATIONOFFICER_GROUP_ID=<value goes here>
+AUTH_DISABLED=true
```
Comment (radum) [inline apps/web/.env.development:36]: Copy this one into the prod one and set it to false there.

File: apps/web/src/server/app/auth/auth.controller.js [hunk 3]
```diff
@@ -1,13 +1,25 @@
...
 import { config } from '../../config/config.js';
-import { msalClient } from '../../lib/sso.js';
+import { createMsalClient } from '../../lib/sso.js';
 
 /** @typedef {import('@pins/express').Auth.AuthCodeParams} AuthCodeParams */
 
 const cryptoProvider = new msalNode.CryptoProvider();
 const cryptoUtils = new CryptoUtils();
 
+/** @type {import('@azure/msal-node').ConfidentialClientApplication} */ let msalClient;
```
[the remark this answers is not in the document]
Comment (brpins) [reply apps/web/src/server/app/auth/auth.controller.js:11]: Yeah, sure... I just created it in here as it was the only place msalClient was in use. I can move it into sso.js

File: apps/web/.env.development (hunk 2 above)
Comment (brpins) [reply apps/web/.env.development:46]: Done

File: apps/web/.env.development (hunk 1 above)
Comment (brpins) [reply apps/web/.env.development]: Done

File: apps/web/.env.development (hunk 2 above)
Comment (brpins) [reply apps/web/.env.development:46]: Done

**Pull request merged**